### ISSUE TITLE
Ambassador 0.40.1 floods logs when using add_request_headers with DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT

### ISSUE THREAD

**Issue by ericbarch** [datawire/ambassador#980]
**Describe the bug**
Our logs have begun filling up with `addReferenceKey attempted to add an empty value` entries after upgrading to 0.40.1 (from 0.40.0).

**To Reproduce**
Configure "add_request_headers" with the value "%DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT%" in a service. Make requests to the service.

**Expected behavior**
No errors or spurious logs should be generated.

**Versions (please complete the following information):**
 - Ambassador: 0.40.1
 - Kubernetes environment: Bare metal (bootstrapped via rke).
 - Version: 1.10.5

**Additional context**
Downgrading to 0.40.0 eliminates the log flooding issue.

**Comment by ericbarch** [comment]
Also adding the full log lines here:

```
ambassador-5znxl: [2018-11-15 23:43:47.934][77][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
ambassador-5znxl: [2018-11-15 23:43:48.585][77][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
ambassador-5znxl: [2018-11-15 23:43:52.947][75][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
ambassador-5znxl: [2018-11-15 23:43:52.949][77][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
ambassador-5znxl: [2018-11-15 23:43:53.673][77][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
ambassador-5znxl: [2018-11-15 23:43:53.695][80][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
```

**Comment by gsagula** [comment]
Thanks, I will see with the team what is the priority on that.

`Downgrading to 0.40.0 eliminates the log flooding issue.` 

Keep in mind that 0.40.0 might crash at any time due to the bug fixed in 0.40.1.

**Comment by ericbarch** [comment]
Roger that. We've actually switched back to running 0.40.1 for that reason. It's a bit of a nuisance but better than crashing. 😃

**Comment by samuela** [comment]
I'm also seeing this behavior in 0.40.2:
```
{
 insertId:  "1awkawcfjfm5kt"  
 logName:  "projects/<redacted>/logs/stderr"  
 metadata: {
  systemLabels: {
   container_image:  "quay.io/datawire/ambassador:0.40.2"    
   container_image_id:  "docker-pullable://quay.io/datawire/ambassador@sha256:b6a95724f70f2b66e08a956ade3e16f7785a70d2c0785cf0446dd8494d886c20"    
   name:  "ambassador"    
   node_name:  "gke-<redacted>-pool-1-088258f8-f2nx"    
   provider_instance_id:  "4661639604121429816"    
   provider_resource_type:  "gce_instance"    
   provider_zone:  "us-west2-a"    
   service_name: [
    0:  "ambassador-admin"     
    1:  "ambassador"     
   ]
   top_level_controller_name:  "ambassador"    
   top_level_controller_type:  "Deployment"    
  }
  userLabels: {
   pod-template-hash:  "10592405"    
   service:  "ambassador"    
  }
 }
 receiveTimestamp:  "2018-12-25T01:25:03.862311026Z"  
 resource: {
  labels: {
   cluster_name:  "<redacted>"    
   container_name:  "ambassador"    
   location:  "us-west2"    
   namespace_name:  "ambassador"    
   pod_name:  "ambassador-549f6849-cxvbj"    
   project_id:  "<redacted>"    
  }
  type:  "k8s_container"   
 }
 severity:  "ERROR"  
 textPayload:  "[2018-12-25 01:25:03.703][58][error][config] source/common/http/header_map_impl.cc:373] addReferenceKey attempted to add an empty value
"  
 timestamp:  "2018-12-25T01:25:03Z"  
}
```

**Comment by richarddli** [comment]
This should be addressed in 0.50.x. Please reopen if that's not the case.
